config/routes.rb

```diff
@@ -450,15 +450,18 @@
     get "/feed/:username", to: "articles#feed", as: "user_feed", defaults: { format: "rss" }
     get "/rss", to: "articles#feed", defaults: { format: "rss" }
 
-    get "/tag/:tag", to: "stories#index"
-    get "/t/:tag", to: "stories#index", as: :tag
+    namespace :stories do
+      get "/tag/:tag", to: "tagged_articles#index"
+      get "/t/:tag", to: "tagged_articles#index", as: :tag
+      get "/t/:tag/top/:timeframe", to: "tagged_articles#index"
+      get "/t/:tag/page/:page", to: "tagged_articles#index"
+      get "/t/:tag/:timeframe", to: "tagged_articles#index",
+                                constraints: { timeframe: /latest/ }
+    end
```

Comment:
**blocking**: 

We can't change URL as this is going to break all existing tag related URLs (and likely some tests as well).

Can you revert this part and just change where they point to? 

Instead of `to: "stories#index"` you can use `to: "stories/tagged_articles#index"`

Reply:
Sorry for the uncautious change, I have updated it please check it 😄